2-api/server.js

```diff
@@ -0,0 +1,275 @@
+  pool.query(
+    "SELECT * FROM customers WHERE id=$1",
+    [customerId],
+    (err, result) => {
+      if (!err) {
+        res.status(200).json(result.rows);
+      } else {
+        res.status(400);
+        console.error(err);
+      }
+    }
+  );
+});
+
+app.get("/customers/:customerId/orders", (req, res) => {
+  const customerId = req.params.customerId;
+  pool
+    .query(
+      "SELECT orders.order_reference, orders.order_date, products.product_name, product_availability.unit_price, supplier.supplier_name, order_items.quantity FROM order_items INNER JOIN orders on order_id = orders.id INNER JOIN customers ON customer_id = customers.id INNER JOIN product_availability ON prod_id=product_id INNER JOIN suppliers ON supp_id = suppliers.id INNER JOIN products ON prod_id = products.id WHERE customers.id=$1",
+      [customerId]
+    )
+    .then((result) => {
+      res.status(200).json(result.rows);
+    })
+    .catch((err) => console.error(err));
+});
+
+app.put("/customers/:customerId", (req, res) => {
+  const customerId = req.params.customerId;
+  const updateName = req.body.name;
+  const updateAddress = req.body.address;
+  const updateCity = req.body.city;
+  const updateCountry = req.body.country;
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400).send("Customer not found");
+      } else {
+        pool.query(
+          `UPDATE customers SET name=$1, address=$2, city=$3, country=$4 WHERE id=${customerId}`,
+          [updateName, updateAddress, updateCity, updateCountry],
+          (err, result) => {
+            if (!err) {
+              res
+                .status(200)
+                .send("Customer successfully updated")
+                .json(result.rows);
+            } else {
+              console.error(err);
+            }
+          }
+        );
+      }
+    });
+});
+
+app.post("/customers/:customerId/orders", (req, res) => {
+  const customerId = req.params.customerId;
+  const newOrderDate = req.body.orderDate;
+  const newOrderRef = req.body.orderRef;
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400);
+      }
+    })
+    .query(
+      "INSERT INTO orders (customer_id, order_date, order_reference) VALUES ($1, $2, $3)",
+      [customerId, newOrderDate, newOrderRef],
+      (err, result) => {
+        if (!err) {
+          res.status(200).json(result.rows);
+        } else {
+          res.status(400);
+          console.error(err);
+        }
+      }
+    );
+});
+
+app.post("/customers", (req, res) => {
+  const newName = req.body.name;
+  const newAddress = req.body.address;
+  const newCity = req.body.city;
+  const newCountry = req.body.country;
+  if (!newName) {
+    return res.status(404).send("Please enter your name to proceed");
```

Comment:
Good check, and you have correctly identified that only the name field is mandatory in the database table. Mayne a 400 would be a better response code.